hadoop-tools/hadoop-azure/src/main/java/org/apache/hadoop/fs/azure/security/TokenUtils.java

```diff
@@ -40,10 +39,10 @@ private TokenUtils() {
   public static Token<DelegationTokenIdentifier> toDelegationToken(
       final Map<?, ?> inputMap) throws IOException {
     final Map<?, ?> m = (Map<?, ?>) inputMap.get(Token.class.getSimpleName());
-    return (Token<DelegationTokenIdentifier>) toToken(m);
+    return toToken(m);
   }
 
-  public static Token<? extends TokenIdentifier> toToken(final Map<?, ?> m)
+  public static Token<DelegationTokenIdentifier> toToken(final Map<?, ?> m)
```

Comment:
The only possible return type from this function (apart from null) is Token<DelegationTokenIdentifier>. Therefore it seems safe to deterministically define the return type instead of generic return type here. This will solve the Unchecked Cast Error.